tests/signature.rs

```diff
@@ -0,0 +1,37 @@
+use fuel_crypto::{Hasher, PublicKey, SecretKey};
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+
```

Comment:
Should we have any negative tests? E.g. invalid signatures / bad recovery values. 

I know this is probably covered by our secp256k lib, but it may be a good idea to ensure the errors that arise from that scenario are handled appropriately in our error chain (instead of a panic or unsafe undefined behavior).

Reply:
https://github.com/FuelLabs/fuel-crypto/pull/1/commits/2813f10106c20ef3fd349fcf38482a9e168a231d